core/src/main/java/bisq/core/api/model/PaymentAccountTypeAdapter.java

```diff
@@ -0,0 +1,329 @@
+import com.google.gson.stream.JsonWriter;
+
+import java.io.IOException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Predicate;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static bisq.common.util.ReflectionUtils.getSetterMethodForFieldInClassHierarchy;
+import static bisq.common.util.ReflectionUtils.getVisibilityModifierAsString;
+import static bisq.common.util.ReflectionUtils.isSetterOnClass;
+import static bisq.common.util.ReflectionUtils.loadFieldListForClassHierarchy;
+import static bisq.core.locale.CountryUtil.findCountryByCode;
+import static bisq.core.locale.CurrencyUtil.getCurrencyByCountryCode;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.util.Comparator.comparing;
+
+@Slf4j
+class PaymentAccountTypeAdapter extends TypeAdapter<PaymentAccount> {
+
+    private final Class<? extends PaymentAccount> paymentAccountType;
+    private final Class<? extends PaymentAccountPayload> paymentAccountPayloadType;
+    private final Map<Field, Optional<Method>> fieldSettersMap;
+    private final Predicate<Field> isExcludedField;
+
+    /**
+     * Constructor used when de-serializing a json payment account form into a
+     * PaymentAccount instance.
+     *
+     * @param paymentAccountType the PaymentAccount subclass being instantiated
+     */
+    public PaymentAccountTypeAdapter(Class<? extends PaymentAccount> paymentAccountType) {
+        this(paymentAccountType, new String[]{});
+    }
+
+    /**
+     * Constructor used when serializing a PaymentAccount subclass instance into a json
+     * payment account json form.
+     *
+     * @param paymentAccountType the PaymentAccount subclass being serialized
+     * @param excludedFields a string array of field names to exclude from the serialized
+     *                       payment account json form.
+     */
+    public PaymentAccountTypeAdapter(Class<? extends PaymentAccount> paymentAccountType, String[] excludedFields) {
+        this.paymentAccountType = paymentAccountType;
+        this.paymentAccountPayloadType = getPaymentAccountPayloadType();
+        this.isExcludedField = (f) -> Arrays.stream(excludedFields).anyMatch(e -> e.equals(f.getName()));
+        this.fieldSettersMap = getFieldSetterMap();
+    }
+
+    @Override
+    public void write(JsonWriter out, PaymentAccount account) throws IOException {
+        // We write a blank payment acct form for a payment method id.
+        // We're not serializing a real payment account instance here.
+        if (log.isDebugEnabled())
+            log.debug("Writing PaymentAccount json form for fields with accessors...");
```

Comment:
Why check `if isDebugEnabled()` instead of just `log.debug(...)` without the `if`?

Reply:
It is intended to save unnecessary time in the logging layer, that's why I do this.  In this case, the performance savings is trivially small because the logged message is just a string literal.

For things like 
```
    log.debug("{} blah {} blah blah {} ... {} .. {}", 
       someCheapCalculation1, 
       someCheapCalculation2, 
       someExpensiveCalculation3, 
       literal4, 
       someReallyExpensiveCalculation5);
```

the cheap and expensive calculations are performed even if the log level is ERROR.

If wrapped by `if (log.isDebugEnabled())`, the calculations are skipped.  But I don't like the additional lines either.  And in this case it buys nothing, I'll fix it.

I think there are situations where this is a good thing to do, but this isn't one of them.   There are other places I'll fix for the same reason.

Reply:
Agreed that it's worth doing in certain cases where the log is incurring a significant load and/or is very frequent.

Reply:
Unnecessary `log.isDebugEnabled()` were removed in commit 45cfb952450ddfa50458aca259bf90447c52dab9.